iP/src/duke/Duke.java

```diff
@@ -0,0 +1,172 @@
+package duke;
+import java.io.IOException;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.HashMap;
+
+/**
+ * Duke is the main class in this todo app.
+ */
+public class Duke {
+    private Storage storage;
+    public TaskList tasks;
+    private Ui ui;
+
+    /**
+     * constructor of Duke
+     * @param filePath path where saved data of todo is stored
+     */
+    public Duke(String filePath) {
+        ui = new Ui();
+        storage = new Storage(filePath);
+        try {
+            // need empty tasks to load properly
+            tasks = new TaskList();
+            tasks = new TaskList(storage.load(this));
+        } catch (IOException e) {
+            ui.showLoadingError();
+            tasks = new TaskList();
+        }
+    }
+
+    /**
+     * load tasks from saved text file
+     * takes user input and execute until program terminates
+     */
+    public void run() {
+        ui.showWelcome();
+        boolean isExit = false;
+        while (!isExit) {
+            try {
+                String fullCommand = ui.readCommand();
+                ui.showLine(); // show the divider line ("_______")
+                /*
+                Command c = Parser.parse(fullCommand);
+                c.execute(tasks, ui, storage);
+                isExit = c.isExit();
+                 */
+                user_input_handler(fullCommand, false);
+            } catch (DukeException | IOException e) {
+                ui.showError(e.getMessage());
+            } finally {
+                ui.showLine();
+            }
+        }
+    }
+
+    /**
+     * run todo app
+     */
```

Comment:
Not sure if this would be necessary as it is simple enough fro a reader to understand